### DISCUSSION TITLE
Tanstack Query cache and reloading

### DISCUSSION THREAD

**Discussion by nataliasposts** [TanStack/query#6811] [Q&A]
Hi

Could someone explain me, I call useQuery in the main page. I took this data from query cache on another page and reload this page by hand, in the result I have cache is empty. Why?

**Comment by TkDodo** [comment]
because the cache is in-memory only and doesn't persist between page loads unless you opt in into persisters

**Reply by nataliasposts** [reply]
but if I am navigating within a SPA app it will persist

**Reply by TkDodo** [reply]
yes because that doesn't re-load the page

**Reply by nataliasposts** [reply]
yes, but if I hard reload the page (by hand in the browser) I will lost this data and need to use StoragePersister?

**Reply by TkDodo** [reply]
yes
